### PR TITLE
FIX: `resample_img()` compatibility with non-NIfTI images

### LINKED ISSUE #3461: `resample_img` forces input image to be NIfTI
Originally popped up in https://github.com/nipreps/smriprep/issues/316 - while plotting `MGHImage`s, sometimes the image needs to be resampled, which is failing with:

```python
  File "/opt/conda/lib/python3.9/site-packages/niworkflows/viz/utils.py", line 377, in plot_registration
	  display = plot_anat(anat_nii, **plot_params)
  File "/opt/conda/lib/python3.9/site-packages/nilearn/plotting/img_plotting.py", line 472, in plot_anat
	  display = plot_img(anat_img, cut_coords=cut_coords,
  File "/opt/conda/lib/python3.9/site-packages/nilearn/plotting/img_plotting.py", line 264, in plot_img
	  display = _plot_img_with_bg(
  File "/opt/conda/lib/python3.9/site-packages/nilearn/plotting/img_plotting.py", line 201, in _plot_img_with_bg
	  display.add_overlay(new_img_like(img, data, affine),
  File "/opt/conda/lib/python3.9/site-packages/nilearn/plotting/displays/_slicers.py", line 273, in add_overlay
	  ims = self._map_show(img, type='imshow', threshold=threshold, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/nilearn/plotting/displays/_slicers.py", line 349, in _map_show
	  img = reorder_img(img, resample=resampling_interpolation)
  File "/opt/conda/lib/python3.9/site-packages/nilearn/image/resampling.py", line 740, in reorder_img
	  return resample_img(img, target_affine=target_affine,
  File "/opt/conda/lib/python3.9/site-packages/nilearn/image/resampling.py", line 451, in resample_img
	  sform, sform_code = img.get_sform(coded=True)
  AttributeError: 'MGHImage' object has no attribute 'get_sform'
```

`MGHImage` is still a [niimg-like object](https://nilearn.github.io/stable/manipulating_images/input_output.html#niimg-like-objects), so the check should be loosened.

Nilearn version: 0.10.0

### Expected behavior
Resamples the image

### Actual behavior
Assumes the image is 

I'll open a PR to loosen this restriction.

### PR DESCRIPTION
**Pull request by mgxd**
Closes #3461

This adds a guard to ensure the input image has the `get_sform()` method prior to calling it, and a quick test.

### CODE REVIEW

**Review by ymzayek — COMMENTED** [review]
LGTM @mgxd! Just a PEP8 issue, see suggestion. As a new contributor can you add your name to `doc/changes/names.rst` as well as a changelog entry in `doc/changes/latest.rst`?

**Comment by bthirion** [comment]
Thx !
You'll probably have to merge #3458 when it's OK, because current failure make it hard to detect a bug.

**Comment by RaphaelMeudec** [comment]
Small suggestion to avoid a near duplicate with the test above of yours (`test_resample_input`), we could merge the two tests into a parametrized one:

```python
@pytest.mark.parametrize("ImageType", [Nifti1Image, MGHImage])
def test_should_resample_for_multiple_image_types(ImageType):
    rng = np.random.RandomState(42)
    shape = (3, 2, 5, 2)
    data = rng.randint(0, 10, shape, dtype="int32")
    affine = np.eye(4)
    affine[:3, -1] = 0.5 * np.array(shape[:3])
    img = ImageType(data, affine)

    resample_img(img, target_affine=affine, interpolation='nearest')
```

Note: this removes the writing to disk from `test_resample_input`, as `testing.write_tmp_imgs` does not support MGHImage out of the box, but I'm not sure this is an issue for the test? WDYT @ymzayek?

Thanks again for the PR!

**Comment by ymzayek** [comment]
@RaphaelMeudec from a quick look it seems that the test `test_resample_input` is testing the input as a Path object rather than a nibabel image class object. I also noticed that the affine and target affine are the same so I need to better check what the test is trying to do. I think I wouldn't combine them. But from looking at the test file there might be some refactoring that can be done. This should be taken up in a different issue/PR

File: doc/changes/latest.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@ NEW
 Fixes
 -----
 
+- :func:`~image.resampling.resample_img` compatibility with non-NIfTI images (:gh:`3462` by `Mathias Goncalves`_).
```
Comment (ymzayek) [inline doc/changes/latest.rst:14]:
```suggestion
- :func:`~image.resample_img` compatibility with non-NIfTI images (:gh:`3462` by `Mathias Goncalves`_).
```
To set the hyperlink correctly

**Review by ymzayek — COMMENTED** [review]
@mgxd this is almost ready to merge. Can you just fix the sphinx reference as suggested? I also think the whatsnew entry can be a bit more descriptive. Thx !

**Review by ymzayek — APPROVED** [review]
Thx @mgxd!

File: nilearn/image/resampling.py [hunk 2]
```diff
@@ -448,11 +448,12 @@ def resample_img(img, target_affine=None, target_shape=None,
 
     # If later on we want to impute sform using qform add this condition
     # see : https://github.com/nilearn/nilearn/issues/3168#issuecomment-1159447771 # noqa:E501
-    sform, sform_code = img.get_sform(coded=True)
-    if not sform_code:
-        warnings.warn("The provided image has no sform in its header. "
-                      "Please check the provided file. "
-                      "Results may not be as expected.")
+    if hasattr(img, 'get_sform'):  # NIfTI images only
```
Comment (bthirion) [inline nilearn/image/resampling.py:451]: I'm wondering if we should send a warning if the condition is `False`. @ymzayek WDYT ?

**Review by bthirion — APPROVED** [review]
Thx a lot !

**Comment by ymzayek** [comment]
Ok merging this!

**Pull request merged**